### PR TITLE
validations and item subdocument

### PR DESCRIPTION
**Pull request by gpark31**

### CODE REVIEW

**Comment by matttarantino** [comment]
Thanks for putting this together! I’ll review this tonight since it has
some overlap with my current PR.

File: data/universities.js [hunk 1]
```diff
@@ -106,15 +105,18 @@ async function updateUniversity(name, emailDomain) {
...
   if (!university) {
     throw 'No university with given id exists!';
   }
 
   const deleteUniversity = await universitiesCollection.deleteOne({
-    _id: id
+    _id: ObjectId(id)
   });
 
   if (deleteUniversity.deletedCount == 0) {
```
Comment (matttarantino) [inline data/universities.js:122]: @gpark31 can we revert the changes to this file? Everything was taken care of in https://github.com/dwilliamson5/CS546_WS_final_project/pull/38

File: data/validations/sharedValidations.js [hunk 2]
```diff
@@ -1,16 +1,14 @@
-function checkArgumentLength(args, length) {
-    let argLength = args.length;
```
Comment (matttarantino) [inline data/validations/sharedValidations.js:2]: We may want to keep this so that you dont have to pass in `arguments.length` every time

File: data/universities.js (hunk 1 above)
Comment (gpark31) [reply data/universities.js:122]: all validations as well?

Comment (gpark31) [reply data/universities.js:122]: i made a different commit with just validations for items and data/item fixed and lemme add my seed

**Comment by matttarantino** [comment]
@gpark31 this still has conflicts with what's already merged into master. Please rebase this branch and resolve merge conflicts
